flapison/querystring.py

```diff
@@ -177,11 +177,10 @@ def fields(self):
                 result[key] = [value]
 
         for key, value in result.items():
-            schema = get_schema_from_type(key)
             for obj in value:
```

Comment:
In your changed code, you're always comparing each field to the current resource's schema (`self.schema`). However this isn't correct, because your request can define fields for a different resource.

Consider `GET /articles?include=author&fields[articles]=title,body&fields[people]=name` from the JSON API spec. Here, the last query parameters filters the fields of `people`, even though the main request is for `articles`. The original code handles this by looking up the resource name in the registry (which is fine), but you don't deal with that issue here.